torch/testing/_internal/common_device_type.py

```diff
@@ -368,17 +369,21 @@ def instantiated_test(self, param_kwargs=param_kwargs):
                 # Note: precision is reset after the test is run
                 guard_precision = self.precision
                 guard_rel_tol = self.rel_tol
+                self.logger = TestLogger()
```

Comment:
This is a cool approach

cc @jbschlosser who's been looking at our test architecture recently